pysb/tools/sensitivity_analysis.py

```diff
@@ -0,0 +1,359 @@
+                               cmap=plt.get_cmap(colors), vmin=v_min,
+                               vmax=v_max)
+            sens_ij_nm.append(per_protein1)
+        plt.tight_layout()
+        plt.savefig(os.path.join(out_dir, '{}_subplots.png'.format(savename)),
+                    bbox_inches='tight')
+        plt.close()
+
+        # Create heatmap and boxplot of data
+        plt.figure(figsize=(14, 10))
+        plt.subplots_adjust(hspace=0.1)
+        outer = gridspec.GridSpec(2, 1, width_ratios=[0.4, 1],
+                                  height_ratios=[0.03, 1])
+
+        gs1 = gridspec.GridSpecFromSubplotSpec(1, 1, subplot_spec=outer[0])
+        gs2 = gridspec.GridSpecFromSubplotSpec(2, 1, subplot_spec=outer[1],
+                                               hspace=.35)
+        ax0 = plt.subplot(gs1[0])
+        ax1 = plt.subplot(gs2[0])
+        # create heatmap of sensitivities
+        im = ax1.imshow(self.p_matrix, interpolation='nearest', origin='upper',
+                        cmap=plt.get_cmap(colors), vmin=v_min,
+                        vmax=v_max, extent=[0, self.nm, 0, self.nm])
+
+        shape_label = np.arange(self.n_sam / 2, self.nm, self.n_sam)
+        plt.xticks(shape_label, self.proteins_of_interest, rotation='vertical',
+                   fontsize=12)
+        plt.yticks(shape_label, reversed(self.proteins_of_interest),
+                   fontsize=12)
+        x_ticks = ([i for i in range(0, self.nm, self.n_sam)])
+        ax1.set_xticks(x_ticks, minor=True)
+        ax1.set_yticks(x_ticks, minor=True)
+        plt.grid(True, which='minor', linestyle='--')
+        color_bar = plt.colorbar(im, cax=ax0, orientation='horizontal',
+                                 use_gridspec=True)
+        color_bar.set_label('% change', y=1, labelpad=5)
+        color_bar.ax.xaxis.set_label_position('top')
+        ticks = np.linspace(v_min, v_max, 5, dtype=int)
+        color_bar.set_ticks(ticks)
+        color_bar.ax.set_xticklabels(ticks)
+
+        # create boxplot of single parameter sensitivities
+        ax2 = plt.subplot(gs2[1])
+        ax2.boxplot(sens_ij_nm[::-1], vert=False, labels=None, showfliers=True,
+                    whis='range')
+        ax2.set_xlim(v_min - 2, v_max + 2)
+        ax2.set_xlabel(x_axis_label, fontsize=12)
+        plt.setp(ax2, yticklabels=reversed(self.proteins_of_interest))
+        ax2.yaxis.tick_left()
+        ax2.set_aspect(1. / ax2.get_data_ratio(), adjustable='box', )
+        plt.savefig(os.path.join(out_dir, savename + '.png'),
+                    bbox_inches='tight')
+        plt.savefig(os.path.join(out_dir, savename + '.eps'),
+                    bbox_inches='tight')
+        plt.savefig(os.path.join(out_dir, savename + '.svg'),
+                    bbox_inches='tight')
+        plt.show()
+
+
+def cartesian_product(array_1, array_2):
+    """
+
+    :param array_1:
+    :param array_2:
+    :return:
+    """
+    a = list(product(array_1, array_2))
```

Comment:
Could potentially do away with this function and iterate over the output of `itertools.product()` directly, since (I think) all calls to this function iterate over the resulting matrix anyway. As `itertools.product()` is a generator, this approach would save memory where the input arrays are large (no intermediate data structure created). If memory isn't an issue in practice then don't worry about it.